packages/components/src/date-time/date.js

```diff
@@ -1,135 +1,119 @@
 /**
  * External dependencies
  */
-import moment from 'moment';
-// react-dates doesn't tree-shake correctly, so we import from the individual
-// component here, to avoid including too much of the library
-import DayPickerSingleDateController from 'react-dates/lib/components/DayPickerSingleDateController';
+import ReactDatePicker, { registerLocale } from 'react-datepicker';
+import { isSameDay, format, getDate } from 'date-fns';
+import { map, filter } from 'lodash';
+import * as locales from 'date-fns/esm/locale';
 
 /**
  * WordPress dependencies
  */
-import { Component, createRef } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import { __experimentalGetSettings as getSettings } from '@wordpress/date';
+
+/**
+ * Internal dependencies
+ */
+import { Button, Tooltip } from '../';
+import { mapLocaleCode } from './utils';
 
 /**
  * Module Constants
  */
 const TIMEZONELESS_FORMAT = 'YYYY-MM-DDTHH:mm:ss';
 const isRTL = () => document.documentElement.dir === 'rtl';
 
-class DatePicker extends Component {
-	constructor() {
-		super( ...arguments );
+/*
+ * Register locale
+ */
+const settings = getSettings();
+const siteLocale = settings?.l10n?.locale || 'en';
+const _l = mapLocaleCode( siteLocale );
+const localeObject = locales[ _l ] ? locales[ _l ] : locales.en_US;
+registerLocale( _l, localeObject );
 
-		this.onChangeMoment = this.onChangeMoment.bind( this );
-		this.nodeRef = createRef();
-		this.keepFocusInside = this.keepFocusInside.bind( this );
-		this.isDayHighlighted = this.isDayHighlighted.bind( this );
-	}
+const renderDatePickerHeader = ( { date, decreaseMonth, increaseMonth } ) => (
+	<div className={ 'components-datetime__date-header' }>
+		<Button
+			className={ `components-datetime__date-header-month-button is-previous-month` }
+			icon={ 'arrow-left-alt' }
+			isSmall={ true }
+			onClick={ decreaseMonth }
+		/>
```

Comment:
This button is empty so screen readers will only announce "button".
Te original react-datepicker button has both off-screen text and an aria-label.

Reply:
yes, agree. Keep in mind Andrea this PR is full of remaining issues. The primary purpose is to be able to know is switching the component will allow us to improve the whole implementation.
Anyway, I'm blad to keep update and polishing!